File: Codeforces Problems/Advantage.java
import java.util.*;

public class Advantage {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int t = sc.nextInt();
        while (t-- > 0) {
            int n = sc.nextInt(), i, max = 0, smax = 0;

            int[] a = new int[n];

            a[0] = sc.nextInt();
            max = a[0];
            smax = a[0];

            for (i = 1; i < n; i++) {
                a[i] = sc.nextInt();

                if (a[i] > max) {
                    smax = max;
                    max = a[i];
                }
                else if (a[i] > smax) {
                    smax = a[i];
                }
            }

            // System.out.println(smax + " " + max);

            for (i = 0; i < n; i++) {
                if (a[i] == max) {
                    System.out.print(a[i] - smax + " ");
                } else {
                    System.out.print(a[i] - max + " ");
                }
            }

            System.out.println();
        }
    }
}

File: Codeforces Problems/advantage.c
#include <stdio.h>
int main()
{
    int t;
    scanf("%d", &t);

    for (int i = 0; i < t; i++)
    {
        int n, max = 0, smax = 0, cmax = 0;
        scanf("%d", &n);

        int a[n];
        for (int j = 0; j < n; j++)
        {
            scanf("%d", &a[j]);
        }

        max = a[0];
        smax = 0;

        for (int j = 0; j < n; j++)
        {
            if (a[j] > max)
            {
                max = a[j];
            }
        }

        for (int j = 0; j < n; j++)
        {
            if (a[j] > smax && a[j] < max)
            {
                smax = a[j];
            }
            if (a[j] == max)
            {
                cmax++;
            }
        }

        if (cmax == 1)
        {
            for (int j = 0; j < n; j++)
            {
                if (a[j] == max)
                {
                    a[j] -= smax;
                }
                else
                {
                    a[j] -= max;
                }
            }
        }
        else
        {
            for (int j = 0; j < n; j++)
            {
                a[j] -= max;
            }
        }

        for (int j = 0; j < n; j++)
        {
            printf("%d ", a[j]);
        }
        printf("\n");
    }
}
